### PR TITLE
TA-1871 remove disability code validation

### PR DESCRIPTION
**Pull request by solomonahmed**
## What does this pull request do?

Please describe what you did.
New feature test to ensure disability code COG can be used after 1st April 2013.  It was previously being blocked by validation which is being removed.

I've also refactored some of the step definitions which create submission lines manually.  Minor mods as well to field/default yaml files to support features/refactoring.

## Why make these changes?

Please describe why the changes were needed.
The business have confirmed that the disability codes COG (& PHY/SEN - although these are not explicitly tested here) should be permitted for Crime and Civil submissions.

Refactoring work done to reduce tech debt and de-couple slightly the step definitions from the feature tests.

## Checklist

- [x] Provided link to a JIRA ticket: https://dsdmoj.atlassian.net/browse/TA-1871

### CODE REVIEW

File: features/step_definitions/cwa_outcome_steps.rb [hunk 1]
```diff
@@ -44,228 +32,42 @@
...
+      area_of_law: area_of_law.downcase.gsub(' ', '_'),
+      category_of_law: category_of_law.downcase.gsub(' ', '_')
+    )
     builder.overrides = test_data_row
     page = AddOutcomePage.new(builder)
     page.add_outcome
   end
 end
 
-When("user adds outcomes for Crime Lower Criminal Proceedings with fields like this:") do |table|
```
Comment (your) [inline features/step_definitions/cwa_outcome_steps.rb:55]: The features under `features/pricing/crime_lower/criminal_proceedings/` need to be updated accordingly

File: features/step_definitions/cwa_outcome_steps.rb [hunk 2]
```diff
@@ -44,228 +32,42 @@
...
+      category_of_law: category_of_law.downcase.gsub(' ', '_')
+    )
+
+    builder.overrides = outcome_data
     page = AddOutcomePage.new(builder)
     page.add_outcome
   end
-end
-
-When ('user adds an outcome for Immigration with {string}, {string}, {string}, {string}, {string}, {string} and {string}') \
```
Comment (your) [inline features/step_definitions/cwa_outcome_steps.rb:73]: This feature needs to be updated accordingly: `features/validations/legal_help/immigration_and_asylum/ecs_care_leaver_validations.feature`

**Review by your — CHANGES_REQUESTED** [review]
Awesome! 🤩
So excited to see developers proactively apply the DRY principle and reduce tech debt. 🎖️ 

I have to request changes only because I believe a few features need to be updated according to the changes (listed in the comments).

File: features/step_definitions/cwa_outcome_steps.rb (hunk 1 above)
Comment (solomonahmed) [reply features/step_definitions/cwa_outcome_steps.rb:55]: Done

File: features/step_definitions/cwa_outcome_steps.rb (hunk 2 above)
Comment (solomonahmed) [reply features/step_definitions/cwa_outcome_steps.rb:73]: Done

**Review by your — APPROVED** [review]
LGTM

**Pull request merged**